apps/opik-backend/src/main/java/com/comet/opik/domain/OpenTelemetryService.java

```diff
@@ -65,4 +120,44 @@ public Mono<Long> parseAndStoreSpans(@NonNull ExportTraceServiceRequest traceReq
                     return spanService.create(spanBatch);
                 }));
     }
+
+    private Map<String, UUID> otelToOpikTraceIdMapper(List<Span> otelSpans, UUID projectId, String workspaceId) {
+        // checks Redis for the otel traceIds in the batch; have we seen them before?
+        // maps (base64 otel id -> UUIDv7 opik id)
+        final Map<String, UUID> traceIdMapper = new HashMap<>();
+
+        otelSpans.forEach(otelSpan -> {
+            var otelTraceId = otelSpan.getTraceId();
+            var otelTraceIdBase64 = base64OtelId(otelTraceId);
+
+            // do we know this traceId? if we do, skip step
+            if (traceIdMapper.containsKey(otelTraceIdBase64)) {
+                return;
+            }
+
+            // checks if this key is mapped in redis
+            var otelTraceIdRedisKey = redisKey(workspaceId, projectId, otelTraceIdBase64);
+            var optOpikTraceId = Optional
+                    .ofNullable((String) redisson.getBucket(otelTraceIdRedisKey).getAndExpire(REDIS_TTL).block());
```

Comment:
Could we move the Redis logic to another service? This would simplify this service since it seems to have too many responsibilities now.